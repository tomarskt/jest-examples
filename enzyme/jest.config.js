module.exports = {
    "verbose": false,
    "bail": true,
    "testEnvironment": "jsdom",
    "setupFiles": ["<rootDir>/setup/shim.js", "<rootDir>/setup/index.js"],
    "snapshotSerializers": ["enzyme-to-json/serializer"]
}