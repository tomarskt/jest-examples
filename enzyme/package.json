{
  "scripts": {
    "test": "jest",
    "debug": "node --inspect-brk node_modules/.bin/jest  --runInBand  "
  },
  "dependencies": {
    "react": "^16.4.0",
    "react-dom": "^16.4.0",
    "prop-types": "^15.6.2"
  },
  "devDependencies": {
    "babel-plugin-transform-async-to-generator": "^6.24.1",
    "babel-plugin-transform-class-properties": "^6.24.1",
    "babel-plugin-transform-object-rest-spread": "^6.26.0",
    "babel-polyfill": "^6.26.0",
    "babel-preset-es2015": "*",
    "babel-preset-react": "*",
    "enzyme": "^3.3.0",
    "enzyme-adapter-react-16": "^1.0.1",
    "enzyme-to-json": "^3.3.4",
    "jest": "^23.4.1",
    "react-router-dom": "^4.3.1",
    "react-test-renderer": "^16.4.0",
    "redux-form": "^7.3.0",
    "lerna-test-library": "^1.2.1"
  }
}
