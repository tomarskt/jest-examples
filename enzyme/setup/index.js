import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// to supress all warnings, uncomment below line;
// no need to install jsdom or setup jsom, as it's part of jest and 
// jest default to jsdom in testEnvironmenet settins. see package.json jest section
// process.env.NODE_ENV = 'production' 


Enzyme.configure({ adapter: new Adapter() });

