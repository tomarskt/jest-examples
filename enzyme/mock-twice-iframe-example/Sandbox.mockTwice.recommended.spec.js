import React from "react";
import { mount } from "enzyme";
import SandBox from "./Sandbox";


jest.mock('../../helpers/dateTimeHelper', () => ({
  getTodayMoment: () => {
    const moment = require('moment');  // do your dynamic require here;
    return moment('2019-04-03');
  },
}));

// babel-plugin-jest-hoist: The module factory of `jest.mock()` is not allowed to reference any out-of-scope variables.
let mockComponent = <div>initial value</div>;
jest.mock("./displayloader", () => () => mockComponent); //// https://github.com/facebook/jest/issues/2567#issuecomment-272015844

describe("recommended: mock twice", () => {
  it("using version ONE", () => {
    mockComponent = <div>Version One</div>;
    const mounted = mount(<SandBox url="http://unit-test.com" />);
    expect(mounted).toMatchSnapshot();
  });

  it("using version TWO", () => {
    mockComponent = <div>Version TWO</div>;
    const mounted = mount(<SandBox url="http://unit-test.com" />);
    expect(mounted).toMatchSnapshot();
  });
});
