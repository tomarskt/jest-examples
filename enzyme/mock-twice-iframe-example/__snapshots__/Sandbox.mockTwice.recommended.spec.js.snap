// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recommended: mock twice using version ONE 1`] = `
<Sandbox
  url="http://unit-test.com"
>
  <Component>
    <div>
      Version One
    </div>
  </Component>
  <GelXIframe
    displayLoader={<Unknown />}
    postMessage={[Function]}
    url="http://unit-test.com"
  >
    <div>
      <Component>
        <div>
          Version One
        </div>
      </Component>
      <div>
        mocked ver :iframe source =
        http://unit-test.com
      </div>
    </div>
  </GelXIframe>
</Sandbox>
`;

exports[`recommended: mock twice using version TWO 1`] = `
<Sandbox
  url="http://unit-test.com"
>
  <Component>
    <div>
      Version TWO
    </div>
  </Component>
  <GelXIframe
    displayLoader={<Unknown />}
    postMessage={[Function]}
    url="http://unit-test.com"
  >
    <div>
      <Component>
        <div>
          Version TWO
        </div>
      </Component>
      <div>
        mocked ver :iframe source =
        http://unit-test.com
      </div>
    </div>
  </GelXIframe>
</Sandbox>
`;
